static/app/components/smartSearchBar/searchDropdown.tsx

```diff
@@ -78,21 +126,102 @@ class SearchDropdown extends PureComponent<Props> {
     </SearchDropdownGroup>
   );
 
-  renderItem = (item: SearchItem) => (
-    <SearchListItem
-      key={item.value || item.desc || item.title}
-      className={item.active ? 'active' : undefined}
-      data-test-id="search-autocomplete-item"
-      onClick={item.callback ?? this.props.onClick.bind(this, item.value, item)}
-      ref={element => item.active && element?.scrollIntoView?.({block: 'nearest'})}
-    >
-      <SearchItemTitleWrapper>
-        {item.title && `${item.title}${item.desc ? ' · ' : ''}`}
-        <Description>{this.renderDescription(item)}</Description>
-        <Documentation>{item.documentation}</Documentation>
-      </SearchItemTitleWrapper>
-    </SearchListItem>
-  );
+  renderQueryItem = (item: SearchItem) => {
+    if (!item.value) {
+      return null;
+    }
+
+    const parsedQuery = parseSearch(item.value);
+
+    if (!parsedQuery) {
+      return null;
+    }
+
+    return (
+      <QueryItemWrapper>
+        <HighlightQuery parsedQuery={parsedQuery} />
+      </QueryItemWrapper>
+    );
+  };
+
+  renderItem = (item: SearchItem, isChild?: boolean) => {
```

Comment:
`renderItem` also seems like it can be extracted to its own component. And the logic for creating `children` can also be extracted.